pyonepassword/op_items/_new_item.py

```diff
@@ -132,49 +134,72 @@ def __init__(self, title: str, fields: List[OPItemField] = [], sections: List[OP
+        # things up, but since we do have to check first, lets just raise
+        # a descriptive OPInvalidItemException
+        if hasattr(self, "CATEGORY"):
+            template_dict: Dict = directory.template_for_category(
+                self.CATEGORY)
+        else:
+            raise OPInvalidItemException(
+                "Item template class inherit also from inherit a concreate OPAbstractItem implementation, overriding CATEGORY")
         template_dict["title"] = title
         section_map = {}
         new_sections = []
         new_fields = []
         old_to_new_sections = {}
-        for section in sections:
-            if not isinstance(section, OPNewSection):
-                old_id = section.section_id
-                section = OPNewSection.from_section(section)
+
+        for sect in sections:
+            if not isinstance(sect, OPNewSection):
+                old_id = sect.section_id
+                sect = OPNewSection.from_section(sect)
                 # create a mapping of old-to-new sections to keep
                 # up with the sections whose UUIDs get regenerated
-                old_to_new_sections[old_id] = section
-            if section.section_id in section_map:
+                old_to_new_sections[old_id] = sect
+            if sect.section_id in section_map:
                 raise OPSectionCollisionException(
-                    f"Duplicate section ID when creating new sections: {section.section_id}")
-            section_map[section.section_id] = section
-            new_sections.append(section)
+                    f"Duplicate section ID when creating new sections: {sect.section_id}")
+            section_map[sect.section_id] = sect
+            new_sections.append(sect)
 
         for field in fields:
             section_id = field.section_id
             if not isinstance(field, OPNewItemField):
-                section = old_to_new_sections.get(section_id)
-                field = OPNewItemFieldFactory.item_field(field, section)
+                section = None
+                if section_id:
+                    section = old_to_new_sections.get(section_id)
+                field = OPNewItemFieldFactory.item_field(
+                    field, section=section)
             else:
-                section = old_to_new_sections.get(section_id)
+
+                section = old_to_new_sections.get(
+                    section_id) if section_id else None
                 if section:
                     # If a section's UUID may have been regenerated, we need to re-link it to the field
-                    field.update_section(section)
+                    field.update_section(sect)
             new_fields.append(field)
         template_dict["sections"] = new_sections
         template_dict["fields"] = new_fields
         key_collisions = []
+
         for key in extra_data.keys():
-            if key in self:
+            if key in self:  # type: ignore
```

Comment:
does it complain because this is a mixin and so the mixin itself doesn't guarantee that the class it's mixed into is iterable?

You can do something like 

```
from collections.abc import Iterable

if isinstance(self, Iterable):
    ...
```

To see if this is indeed iterable

Reply:
Yes that's correct. I decided to leave it as-is though, because it was always my intention for this to be a hard failure, and blow up as-is if the mixin isn't used properly